src/lib/SoftHSM.cpp

```diff
@@ -670,6 +670,9 @@ CK_RV SoftHSM::C_GetMechanismList(CK_SLOT_ID slotID, CK_MECHANISM_TYPE_PTR pMech
 		CKM_SHA256_RSA_PKCS,
 		CKM_SHA384_RSA_PKCS,
 		CKM_SHA512_RSA_PKCS,
+#ifdef WITH_OPENSSL
```

Comment:
You also need to increase the number of supported mechanisms above. E.g:

```
#ifdef WITH_OPENSSL
 	nrSupportedMechanisms += 1;
#endif
```

Reply:
Thanks. Included in the updated patch.